dashboard/cron.py

```diff
@@ -268,4 +81,339 @@ def util_function(sql_string, mysql_table, table_identifier=None):
+    logger.debug("in update_cron_status")
+
+    engine = create_engine("mysql+pymysql://{user}:{password}@{host}:{port}/{db}"
+                           .format(db = db_name,  # your mysql database name
+                                   user = db_user, # your mysql user for the database
+                                   password = db_password, # password for user
+                                   host = db_host,
+                                   port = db_port))
+    connection = engine.connect()
+    sql = "insert into cron_status (cron_date, status) values('" + str(datetime.datetime.now()) + "', '" + status + "')"
+    logger.debug("sql=" + sql)
+    connection.execute(sql)
+    connection.close()
+
+    logger.info("updated cron_status")
+
+# remove all records inside the specified table
+def deleteAllRecordInTable(tableName):
+    # delete all records in the table first
+    engine = create_engine("mysql+pymysql://{user}:{password}@{host}:{port}/{db}"
+                           .format(db = db_name,  # your mysql database name
+                                   user = db_user, # your mysql user for the database
+                                   password = db_password, # password for user
+                                   host = db_host,
+                                   port = db_port))
+    connection = engine.connect()
+    connection.execute("delete from " + tableName)
+    connection.close()
+
+    return "records removed from " + tableName + ";"
+
+# cron job to populate course and user tables
+class DashboardCronJob(CronJobBase):
+
+    schedule = Schedule(run_every_mins=settings.CRON_RUN_SCHEDULE)
+    code = 'dashboard.DashboardCronJob'    # a unique code
+
+    # update FILE records from UDW
+    def update_with_udw_course(self):
+        # cron status
+        status = ""
+
+        logger.debug("in update with udw course")
+
+        # delete all records in the table first
+        status += deleteAllRecordInTable("course")
+
+        # loop through multiple course ids
+        for UDW_course_id in settings.DEFAULT_UDW_COURSE_IDS:
+            logger.debug("UDW_course_id = " + UDW_course_id)
+
+            #select file record from UDW
+            course_sql = "select id, name, " + settings.CURRENT_CANVAS_TERM_ID + " as term_id from course_dim where id='" + UDW_course_id + "'"
+
+            logger.debug(course_sql)
+
+            status += util_function(UDW_course_id, course_sql, 'course')
+
+        return status
+
+    # update USER records from UDW
+    def update_with_udw_user(self):
+
+        # cron status
+        status = ""
+
+        logger.debug("in update with udw user")
+
+        # delete all records in the table first
+        status += deleteAllRecordInTable("user")
+
+        # loop through multiple course ids
+        for UDW_course_id in settings.DEFAULT_UDW_COURSE_IDS:
+
+            # select all student registered for the course
+            user_sql = "select u.name AS name, " \
```

Comment:
When I was writing these queries for validation

putting them in """ ... """ made the queries cleaner. https://github.com/tl-its-umich-edu/unizin-sis-validation/blob/master/dbqueries.py

You can use the regular string formatting with % or .format to insert the variables afterward.